### PR TITLE
alts: parameterized ip version for tests

### PR DESCRIPTION
**Pull request by botengyao**
Parameterize the tests, and there is a room to refactor, but haven't got time to do it.

Commit Message:
Additional Description:
Risk Level: low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: test/extensions/transport_sockets/alts/tsi_handshaker_test.cc [hunk 1]
```diff
@@ -176,7 +181,8 @@ class AltsTsiHandshakerTest : public Test {
       server_ = server_builder.BuildAndStart();
       EXPECT_THAT(server_, NotNull());
       EXPECT_NE(listening_port, -1);
-      server_address_ = absl::StrCat("[::1]:", listening_port);
+      server_address_ =
+          absl::StrCat(Network::Test::getLoopbackAddressUrlString(version_), ":", listening_port);
```
Comment (tyxia) [inline test/extensions/transport_sockets/alts/tsi_handshaker_test.cc:185]: Looks like there is build error.  `version_` is in test base `HttpIntegrationTest` or base_integration_test.h

**Review by tyxia — COMMENTED** [review]
Thanks for prompt fix

File: test/extensions/transport_sockets/alts/tsi_handshaker_test.cc (hunk 1 above)
Comment (botengyao) [reply test/extensions/transport_sockets/alts/tsi_handshaker_test.cc:185]: Right, most of the tests are not inheriting `HttpIntegrationTest`.

**Comment by tyxia** [comment]
cced @matthewstevenson88

**Comment by matthewstevenson88** [comment]
Thanks @botengyao!!

**Review by KBaichoo — APPROVED** [review]

**Pull request merged**